src/libraries/System.IO.FileSystem/tests/File/ReadWriteAllLinesAsync.cs

```diff
@@ -199,4 +199,103 @@ public override Task TaskAlreadyCanceledAsync()
                 async () => await File.WriteAllLinesAsync(path, new[] { "" }, Encoding.UTF8, token));
         }
     }
+
+    public class File_ReadLinesAsync : File_ReadWriteAllLines_EnumerableAsync
+    {
+        protected override Task<string[]> ReadAsync(string path)
+        {
+            return ReadAsync(path, default, default);
+        }
+
+        [Fact]
+        public override async Task DisposingEnumeratorClosesFileAsync()
+        {
+            string path = GetTestFilePath();
+            await WriteAsync(path, new[] { "line1", "line2", "line3" });
+
+            IAsyncEnumerable<string> readLines = File.ReadLinesAsync(path);
+            await using (IAsyncEnumerator<string> e1 = readLines.GetAsyncEnumerator())
+            await using (IAsyncEnumerator<string> e2 = readLines.GetAsyncEnumerator())
+            {
+                Assert.Same(readLines, e1);
+                Assert.NotSame(e1, e2);
+
+                await e1.MoveNextAsync();
+            }
+
+            // File should be closed deterministically; this shouldn't throw.
+            await File.OpenWrite(path).DisposeAsync();
+        }
+
+        [Fact]
+        public override async Task TaskAlreadyCanceledAsync()
+        {
+            string path = GetTestFilePath();
+            File.Create(path).Dispose();
+            var ct = new CancellationToken(true);
+            await Assert.ThrowsAsync<TaskCanceledException>(() => ReadAsync(path, ct, default));
+            await Assert.ThrowsAsync<TaskCanceledException>(() => ReadAsync(path, default, ct));
+        }
+
+        [Fact]
+        public virtual void InvalidArgumentsThrowsBeforeGetAsyncEnumerator()
+        {
+            Assert.Throws<ArgumentNullException>("path", () => File.ReadLinesAsync(null));
+            Assert.Throws<ArgumentException>("path", () => File.ReadLinesAsync(string.Empty));
+        }
+
+        protected virtual async Task<string[]> ReadAsync(string path, CancellationToken enumerableCt, CancellationToken enumeratorCt)
+        {
+            var list = new List<string>();
+            await foreach (string item in File.ReadLinesAsync(path, enumerableCt).WithCancellation(enumeratorCt))
+            {
+                list.Add(item);
+            }
+
+            return list.ToArray();
+        }
+    }
+
+    public class File_ReadLines_EncodedAsync : File_ReadLinesAsync
+    {
+        [Fact]
+        public override void InvalidArgumentsThrowsBeforeGetAsyncEnumerator()
+        {
+            string path = GetTestFilePath();
+            Assert.Throws<ArgumentNullException>(() => File.ReadLinesAsync(path, null));
```

Comment:
we should verify the name of the encoding argument as well
```suggestion
            Assert.Throws<ArgumentNullException>("encoding", () => File.ReadLinesAsync(path, null));
```

Reply:
Fixed, thanks.